src/views/README.md

```diff
@@ -0,0 +1,159 @@
+   - These values are referenced in the template in single quotation marks, e.g.:
+    ```
+    {% set errorTitle = 'pages.pageMultipleDocCheck.content.formErrorMessage.errorSummaryTitleText' | translate %}
+    ```
+   - `translate` & variants are defined in `src/config/nunjucks.js`.
+     - It takes the reference as a key to find the content saved in `src/locales` files for english/ welsh.
+     - It can take secondary params, e.g.: `context`, which adds a suffix to the reference it's going to find.
+3. `locales.js`:
+    - the `res.locals` variable attributes are provided as variables in the Nunjucks file, to be accessible like `"{{contactUsUrl}}"`.
+
+## Creating a new page
+
+Assuming this page wants a radio input on it
+
+1. Add a Nunjucks template in `src/views/ipv/page`, e.g. `prove-identity-no-other-photo-id`
+
+Breaking this example down:
+- Needed as contains the common stylesheets, layout.
+    ```html
+    {% extends "shared/base.njk" %}
+    ```
+  - Relevant details:
+    - `pageTitleKey` is used for finding title from content, and `pageErrorState` may trigger an "Error: " prefix.
+    `showLanguageToggle` toggles the language toggle.
+    - `showBack` toggles the back link.
+    - `errorState` toggles error summary, which includes: `errorTitle`, `errorText` and `errorHref`.
+    - `googleTagManagerPageId` sets state.
+
+- Import common, standard GovUK components.
+    ```html
+    {% from "govuk/components/button/macro.njk" import govukButton %}
+    {% from "govuk/components/radios/macro.njk" import govukRadios %}
+    ```
+
+- Sets the page title using English/ Welsh content.
+    ```html
+    {% set pageTitleKey = 'pages.proveIdentityNoOtherPhotoId.title' %}
+    ```
+
+- Sets the tag for analytics.
+    ```html
+    {% set googleTagManagerPageId = "proveIdentityNoOtherPhotoId" %}
+    ```
+
+- Variables for indicating errors, used in shared/base.njk as well.
+    ```html
+    {% set errorState = pageErrorState %}
+    {% set errorTitle = 'pages.proveIdentityNoOtherPhotoId.content.formErrorMessage.errorSummaryTitleText' | translate %}
+    {% set errorText = 'pages.proveIdentityNoOtherPhotoId.content.formErrorMessage.errorSummaryDescriptionText' | translate %}
+    {% set errorHref = "#proveIdentityNoOtherPhotoIdForm" %}
+    ```
+  - `errorHref` should reference the form's id, so it hash-links there.
+
+- The form which POSTs to the same URL as the page.
+    ```html
+    <form id="proveIdentityNoOtherPhotoIdForm" action="/ipv/page/{{ pageId }}" method="POST">
+    ```
+
+- Hidden input for carrying the token from the GET request to the POST request.
+    ```html
+    <input type="hidden" name="_csrf" value="{{ csrfToken }}">
+    ```
+  - `req.csrfToken()` creates the token in the GET route handler
+  - The result is saved in here
+  - And `csrfProtection()` checks it, in the POST route handler.
+  - Arguably not a required security feature since our forms only lead to change in position on a journey map.
```

Comment:
Could we remove this - it's a little subjective and we do have a requirement to protect all our forms to prevent user actions being taken by someone/something malicious even if it is only navigating them to the next page

Reply:
Ah thank you for the clarification. Will remove appreciate it's a bit subjective